Backend Code Cleanup

ronit14235 opened the pull request:
# Pull Request Template

---

## Please check if the PR fulfills these requirements:
- [x] The commit message follows our guidelines.
- [x] Tests for the changes have been added (for bug fixes / features).
- [x] Docs have been added / updated (for bug fixes / features).

---

## What kind of change does this PR introduce?  
This PR includes a **bug fix** and **code cleanup** for the backend and unit tests.

---

## What is the current behavior?  
### Current Issues:
- The timezone handling in the `calendarGoogle.py` module is inconsistent, leading to potential errors when fetching or displaying events.
- Backend and test files contain some unoptimized or unclear code, reducing maintainability.
- Lack of explicit instructions for running the tests within the CI/CD pipeline context.

---

## What is the new behavior?  
### Resolutions and Updates:
- **Bug Fix:** The timezone handling in `calendarGoogle.py` has been resolved, ensuring consistent and correct behavior when managing calendar events.
- **Code Cleanup:** Backend code and unit test files have been refactored for improved readability and maintainability.
- **Testing Instructions:** Clear steps for running unit tests have been provided for both local environments and CI/CD pipelines.

---

## How can we run this change or implement this behavior?

### To verify these changes, run the unit tests as follows:

1. **For Python Unit Tests:**
   - Run the following command from the root directory:
     ```bash
     coverage run -m unittest discover -s tests
     ```

2. **For JavaScript Unit Tests (Jest):**
   - Run the following command from the root directory:
     ```bash
     npm test
     ```
---

myrmanshelby commented on static/js/app.js:
```diff
@@ -104,28 +149,40 @@ async function deleteEvent(eventId) {
   document.getElementById('event-location').value = event.location || '';
   document.getElementById('event-description').value = event.description || '';
-
+  const timeZone = event.start.timeZone || 'UTC';
```
Do we need to assign this as a variable or can we just run event.start.timeZone? Just wondering since it's throwing a codacy error, but it could be a mistake on codacy's end

ronit14235 replied:
Yes the line should have been removed. I forgot. Thank you for pointing it out

myrmanshelby reviewed (approved):
Looks excellent!!! I see there's a new codacy error, so maybe double check that before pushing. Could be a false error though so no pressure to fix it

ronit14235 commented:
### I've fixed the Codacy issue and also ensured that deadlines update without requiring a page refresh.

### Bug Fix: Dynamic Update for Today's Deadlines

---

#### **Bug Fixed:**
- **Issue Fixed:**  
  Resolved the problem where new events or internships added with today's date were not immediately appearing in the "Today's Deadlines" section. Previously, this required a page refresh to reflect the updates.

---

#### **Updates:**
- Fixed the Codacy issue in the codebase.
- The "Today's Deadlines" section now updates dynamically without requiring a page reload whenever new events or internships with today's date are added.

---

This also fixes #76

myrmanshelby reviewed (approved):
looks good!
